example/function/src/goodbye.php

```diff
@@ -5,7 +5,9 @@
 
 function goodbye($data)
 {
-    return "Goodbye, {$data['name']}!";
+    $response = [
+        'msg' => "Goodbye, {$data['name']}!",
+        'data' => $data
+    ];
+    return $response;
 }
-
-?>
```

Comment:
I am a PHP noob, but doesn't the `<?php` at the beginning of the file need to be closed here?

Reply:
It does not :) Open tags are usually necessary but close tags are optional and actually detrimental in a lot of instances (e.g. they can cause extra whitespace in API responses), so unless it's explicitly needed it's best to omit it.

> If a file contains only PHP code, it is preferable to omit the PHP closing tag at the end of the file [PHP tags](https://www.php.net/manual/en/language.basic-syntax.phptags.php)

Reply:
Ah ha, thanks!